### ISSUE TITLE
tf.losses.mean_squared_error is actually sum of squares

### ISSUE THREAD

**Issue by ppwwyyxx** [tensorflow/tensorflow#15056]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: v1.3.0-rc1-5211-gab0fcac 1.5.0-dev20171124
- **Python version**: 3.6
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

To truly get mean squared error one has to explictly use `reduction=Reduction.MEAN`. This indicates bad naming. Maybe `squared_error` is a better name, also similar to other names in `tf.losses`

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue? Please update the label and/or status accordingly.

**Comment by ppwwyyxx** [comment]
My bad. `tf.losses.mean_squared_error` is indeed __mean__ square error. The bad naming is actually `Reduction.SUM_BY_NONZERO_WEIGHTS`. I thought it is a sum but this reduction method is actually an average over nonzero weights.
